cpp/dolfinx/common/utils.cpp

```diff
@@ -0,0 +1,96 @@
+// Copyright (C) 2021 Joseph P. Dean
+//
+// This file is part of DOLFINx (https://www.fenicsproject.org)
+//
+// SPDX-License-Identifier:    LGPL-3.0-or-later
+
+#include "utils.h"
+#include <dolfinx/common/MPI.h>
+#include <dolfinx/graph/AdjacencyList.h>
+#include <iostream>
+#include <sstream>
+#include <string>
+#include <vector>
+
+std::vector<int32_t> dolfinx::common::compute_owned_indices(
+    const xtl::span<const std::int32_t>& indices,
+    const dolfinx::common::IndexMap& index_map)
+{
+  // Split indices into those owned by this process and those that
+  // are ghosts. `ghost_indices` contains the position of the ghost
+  // in index_map->ghosts()
+  std::vector<std::int32_t> owned;
+  std::vector<std::int32_t> ghost_indices;
+  for (std::size_t i = 0; i < indices.size(); ++i)
```

Comment:
Possible to implement this using std::copy/std::copy_if?

Reply:
I don't think so, but I've used `std::for_each` instead